src/Filters/remez_fir.jl

```diff
@@ -380,33 +353,134 @@ plot(f, 20*log10.(abs.(freqz(b2,f,1.0))))
-    desired = convert(Vector{Float64}, desired)
-    weight = convert(Vector{Float64}, weight)
-    
-    grid, des, wt = build_grid(numtaps, bands, desired, weight, grid_density, filter_type)
+    length(bands) == 2length(desired) ||
+        throw(ArgumentError("`desired` must be half the length of `bands`."))
+    length(bands) == 2length(weight) ||
+        throw(ArgumentError("`weight` must be half the length of `bands`."))
+    band_ranges = [(bands[i], bands[i+1]) for i in 1:2:length(bands)]
+    if filter_type == filter_type_differentiator
+        eff = [f -> d*f/Hz for d in desired]
+        wate = [d > 0.0001 ? f -> w/(f/Hz) : f -> w for (w, d) in zip(weight, desired)]
+    else
+        eff = desired
+        wate = weight
+    end
+    band_defs = [r => (d, w) for (r, d, w) in zip(band_ranges, eff, wate)]
+    neg = filter_type in (filter_type_hilbert, filter_type_differentiator)
+    return remez(numtaps, band_defs; Hz=Hz, neg=neg, kwargs...)
+end
+
+"""
+    remez(numtaps::Integer, band_defs;
+          Hz::Real=1.0,
+          neg::Bool=false,
+          maxiter::Integer=25,
+          grid_density::Integer=16)
+
+Calculate the minimax optimal filter using the Remez exchange algorithm [^McClellan1973a] [^McClellan1973b].
+
+This is the simplified API that accepts just 2 required arguments (numtaps, band_defs).
+For a scipy compatible version see the 3 arguments version (numtaps, bands, desired). 
+
+Calculate the filter-coefficients for the finite impulse response
+(FIR) filter whose transfer function minimizes the maximum error
+between the desired gain and the realized gain in the specified
+frequency bands using the Remez exchange algorithm.
+
+# Arguments
+- `numtaps::Integer`: The desired number of taps in the filter. 
+    The number of taps is the number of terms in the filter, or the filter 
+    order plus one.
+- `bands_defs`: A sequence of band definitions.
+    This sequence defines the bands. Each entry is a pair. The pair's
+    first item is a tuple of band edges (low, high). The pair's second item 
+    defines the desired response and weight in that band. The weight is optional
+    and defaults to 1.0. Both the desired response and weight may be either scalars
+    or functions. If a function, the function should accept a real frequency and
+    return the real desired response or real weight. Examples:
+      LPF with unity weights. [(0, 0.475) => 1, (0.5, 1.0) => 0].
+      LPF with weight of 2 in the stop band. [(0, 0.475) => (1, 1), (0.5, 1.0) => (0, 2)].
+      BPF with unity weights. [(0, 0.375) => 0, (0.4, 0.5) => 1, (0.525, 1.0) => 0]
+      Hilbert transformer. [(0.1, 0.95) => 1]; neg=true.
+      Differentiator. [(0.01, 0.99) => (f -> f/2, f -> 1/f)]; neg=true.
+- `Hz::Real`: The sampling frequency in Hz. Default is 1.
+- `neg::Bool`: Whether the filter has negative symmetry or not. Default is false.
+    If false, the filter is symmetric. If true, the filter is asymmetric.
```

Comment:
I think "asymmetric" is the wrong term here. Should be "odd-symmetric", if that is a word.

Reply:
is the property that [scipy calls](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.windows.hann.html#scipy.signal.windows.hann) "symmetric" vs "periodic"? In that nomenclature a "symmetric" window has the same number of points on both sides of the peak (and is usually an odd-length window so the peak is on a sample) vs a "periodic" which only includes one of the extrema (so for an even-length periodic window (the common case) there's the peak still falls on a sample). The latter is called periodic because if you consider it to wrap around then the minimum value isn't repeated.

Reply:
No, `neg=true` means that `h[n]=-h[end+1-n]`.

Reply:
Thanks for the comments, I pushed a minor improvement to the "neg" argument description.